lib/adapters/class-vip-enterprise-search.php

```diff
@@ -20,141 +17,112 @@
 class VIP_Enterprise_Search extends Adapter {
 
 	/**
-	 * Add facets to EP query.
-	 * Filters `ep_post_formatted_args`.
+	 * A callback for the ep_valid_response action hook. Parses aggregations
+	 * from the raw Elasticsearch response and adds the buckets to the
+	 * configured aggregations.
 	 *
-	 * @see \ElasticPress\Indexable\Post\Post::format_args() For the `ep_post_formatted_args` filter.
-	 *
-	 * @param array $query Formatted Elasticsearch query.
-	 * @return array
+	 * @param array $response Response from the Elasticsearch server.
 	 */
-	public function add_facets_to_ep_query( $query ): array {
-		// Do we have any facets specified?
-		$searched_facets = get_query_var( 'fs' );
-		if ( empty( $searched_facets ) ) {
-			return $query;
+	public function action__ep_valid_response( $response ): void {
+		if ( ! empty( $response['aggregations'] ) ) {
+			$this->parse_aggregations( $response['aggregations'] );
 		}
+	}
 
-		// Try to get the list of configured facets.
-		$configured_facets = $this->get_facet_config();
-		if ( empty( $configured_facets ) ) {
-			return $query;
+	/**
+	 * A callback for the ep_elasticpress_enabled filter hook. Overrides the
+	 * normal behavior for ElasticPress to determine if it is enabled to allow
+	 * for an empty search string, if allowable by the configuration on this
+	 * adapter.
+	 *
+	 * @param bool     $enabled  Whether ElasticPress is enabled for the query or not.
+	 * @param WP_Query $wp_query The WP_Query being examined.
+	 *
+	 * @return bool Whether ElasticPress should be active for the query or not.
+	 */
+	public function filter__ep_elasticpress_enabled( $enabled, $wp_query ) {
+		if ( $this->get_allow_empty_search()
+			&& $wp_query->is_search()
+			&& $wp_query->is_main_query()
+			&& ! is_admin()
```

Comment:
You could add a note for why this is not enabled in the admin.  Is is just not needed there, or problematic?

Reply:
Honestly, we're focusing on the front-end implementation at the moment, and I wanted to be really specific in what I was targeting for override so I didn't also have to test the potential impacts on the admin. As we move into making this work with ES Admin I'll likely revisit this bit of code.

Reply:
Sure, that's a good call and I can do that.